editor/editor_node.cpp

```diff
@@ -5124,10 +5210,21 @@ void EditorNode::_load_docks_from_config(Ref<ConfigFile> p_layout, const String
 				atidx = k;
 				break;
 			}
+
 			if (atidx == -1) { // Well, it's not anywhere.
 				continue;
 			}
 
+			if (restore_window_on_load && floating_docks_dump.has(name)) {
+				_restore_floating_dock(floating_docks_dump[name], node, atidx);
+				continue;
+			}
+
+			WindowWrapper *wrapper = Object::cast_to<WindowWrapper>(node);
+			if (wrapper) {
+				_dock_floating_close_request(false, wrapper);
+			}
```

Comment:
I assume this is to unfloat docks when changing editor layout?
It doesn't work.

Reply:
The whole editor layout stuff seems to be broken and only functional for dock position and to save the last editor layout and restore on startup. I didn't dig deeper but seems that this function and the `set_window_layout` methods in plugins aren't called.

Reply:
It's not a startup issue though. When you make a dock floating and use Editor -> Editor Layout -> Default, it's supposed to restore all floating docks to their original position. It doesn't happen.

Reply:
Turns out that I'm dumb and that code is wrong. But there is still something wrong in `set_window_layout`. I need to investigate more.